Compiler/SimCode/SimCodeMain.mo

```diff
@@ -143,14 +145,22 @@ protected
   Absyn.ComponentRef a_cref;
   list<String> libPaths;
   tuple<Integer,HashTableExpToIndex.HashTable,list<DAE.Exp>> literals;
+  Option<SimCode.FmiModelStructure> modelStructure;
 algorithm
   System.realtimeTick(ClockIndexes.RT_CLOCK_SIMCODE);
   a_cref := Absyn.pathToCref(className);
   fileDir := CevalScriptBackend.getFileDir(a_cref, p);
   (libs,libPaths,includes, includeDirs, recordDecls, functions, literals) :=
     SimCodeUtil.createFunctions(p, inBackendDAE);
+
   simCode := createSimCode(inBackendDAE, inInitDAE, NONE(), inRemovedInitialEquationLst,
     className, filenamePrefix, fileDir, functions, includes, includeDirs, libs, libPaths, p, SOME(simSettings), recordDecls, literals, Absyn.FUNCTIONARGS({},{}), isFMU=true, FMUVersion=FMUVersion);
+
+  if FMI.isFMIVersion20(FMUVersion) then
+    modelStructure := SimCodeUtil.createFMIModelStructure(inFMIDer, simCode.modelInfo, simCode.crefToSimVarHT);
+    simCode.modelStructure := modelStructure;
+  end if;
```

Comment:
This is not spreading around, since in my point of view this is an fmi part, so I put it to the` function generateModelCodeFMU` and removed it from the general `createSimCode`. Actually I would like to see most parts of `SimCode.SimCode` record creatable with single functions, instead by one big function like createSimCode.